src/lib.rs

```diff
@@ -636,6 +715,46 @@ impl<'a, T, N: Unsigned> KdIndexTreeN<'a, T, N> {
         self.within_radius_by(query, radius, |item, k| item.at(k))
     }
 }
+#[cfg(feature = "rayon")]
+impl<'a, T: Sync, N: Unsigned> KdIndexTreeN<'a, T, N> {
+    pub fn par_build_by<F>(source: &'a [T], compare: F) -> Self
+    where
+        F: Fn(&T, &T, usize) -> Ordering + Copy + Send,
+    {
+        Self {
+            source,
+            kdtree: KdTreeN::par_build_by((0..source.len()).collect(), move |i1, i2, k| {
+                compare(&source[*i1], &source[*i2], k)
+            }),
+        }
+    }
+
+    pub fn par_build_by_key<Key, F>(source: &'a [T], kd_key: F) -> Self
+    where
+        Key: Ord,
+        F: Fn(&T, usize) -> Key + Copy + Send,
+    {
+        Self::par_build_by(source, |item1, item2, k| {
```

Comment:
This function doesn't compile in my environment.
The compiler suggests to add `Sync` trait bound for `F`, because the closure refers `kd_key` as a reference of `F`.

I suggest to add `move` to the closure to move the ownership of `kd_key` to it, as below:
```suggestion
        Self::par_build_by(source, move |item1, item2, k| {
```

The compiler message was:
```
$ rustc --version
rustc 1.51.0 (2fd73fabe 2021-03-23)
$ cargo check --features rayon                                                                                                                    Checking kd-tree v0.2.0 (/home/terada/dev/private/RReverser/kd-tree)
error[E0277]: `F` cannot be shared between threads safely
   --> src/lib.rs:737:9
    |
720 | /     pub fn par_build_by<F>(source: &'a [T], compare: F) -> Self
721 | |     where
722 | |         F: Fn(&T, &T, usize) -> Ordering + Copy + Send,
    | |_______________________________________________________- required by `KdIndexTreeN::<'a, T, N>::par_build_by`
...
737 |           Self::par_build_by(source, |item1, item2, k| {
    |           ^^^^^^^^^^^^^^^^^^ `F` cannot be shared between threads safely
    |
    = note: required because of the requirements on the impl of `Send` for `&F`
    = note: required because it appears within the type `[closure@src/lib.rs:737:36: 739:10]`
help: consider further restricting this bound
    |
735 |         F: Fn(&T, usize) -> Key + Copy + Send + Sync,
    |                                               ^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `kd-tree`

To learn more, run the command again with --verbose.